src/org/pahappa/systems/registrationapp/services/UserService.java

```diff
@@ -1,4 +1,56 @@
 package org.pahappa.systems.registrationapp.services;
 
+import org.pahappa.systems.registrationapp.models.User;
+
+import java.util.Date;
+import java.util.HashSet;
+
 public class UserService {
+    private final HashSet<User> users = new HashSet<>();
+
+    public void registerUser(String firstname, String lastname, String username, Date dateOfBirth){
+        User user = new User();
+        user.setFirstname(firstname);
+        user.setLastname(lastname);
+        user.setUsername(username);
+        user.setDateOfBirth(dateOfBirth);
+        users.add(user);
```

Comment:
Prevent users from registering users with the same username. Since username is going to act as a unique identifier.

Reply:
@fredmugerwa I was planning to make a loop in the view that automatically checks for whether a username is in use, which only loops asking for a different username until a unique one is found.

Reply:
No, have it in the services. For other validations, maybe they will be in the views side.

Reply:
okay then.

Reply:
Completed.